### PR TITLE
early stopping, dropout; approx. errors; parameter structure 

### PR DESCRIPTION
**Pull request by pedroblossbraga**
changes in the NN:
- early stopping (quit training in case the loss is not improved with a certain patience)
- dropout (randomly dropping neurons at a given rate, to prevent overfitting and improve generalization) 
changes in main / structure:
- saving l1 and l2 mean errors in a csv
- setting up the PINN parameters in a dict structure in main to facilitate testing other activation functions, learning rates, etc (now one can easily import the main function or change parameters to test other values and produce an output table)

### CODE REVIEW

File: results/pinns_tf.png
Comment (Haiyun314) [inline results/pinns_tf.png:1]: I think this image is not well-trained

Comment (pedroblossbraga) [reply results/pinns_tf.png:1]: the last image was generated without early stopping, but with dropout, so we can just re-execute without dropout and technically it is the same implementation as before.
Any further suggestions?

**Review by Haiyun314 — APPROVED** [review]
Add an appropriate prefix to the file names being saved to improve clarity and organization, such as drop_out_pinns.png or drop_out_pinns_weights.h5.

File: results/pinns_tf.png
Comment (Haiyun314) [reply results/pinns_tf.png:1]: It would be better to add a meaningful prefix before saving a file for better organization, such as dropout_pinns.png or dropout_pinns.h5. All other changes look good.

**Comment by pedroblossbraga** [comment]
The table ./results/results_{yyyymdd_HHMMSS}.csv shows the parameters used for execution.
